contentcuration/contentcuration/utils/assessment/qti/archive.py

```diff
@@ -0,0 +1,278 @@
+            expected_length=50,  # Default expected length
+            placeholder_text="Enter your answer here",
+        )
+        # Text entry interaction is an inline element, so wrap it in a paragraph tag.
+        interaction_element = P(children=[interaction_element])
+        # prompt is already a list of elements, so just append the interaction to it.
+        prompt.append(interaction_element)
+        interaction = Div(children=prompt)
+
+        correct_values = [
+            Value(value=answer["answer"])
+            for answer in processed_data["answers"]
+            if answer["correct"]
+        ]
+
+        response_declaration = ResponseDeclaration(
+            identifier="RESPONSE",
+            cardinality=Cardinality.MULTIPLE
+            if processed_data["multiple_select"]
+            else Cardinality.SINGLE,
+            base_type=BaseType.STRING,
+            correct_response=CorrectResponse(value=correct_values)
+            if correct_values
+            else None,
+        )
+        return interaction, response_declaration
+
+    def _qti_item_filepath(self, assessment_id):
+        return f"items/{assessment_id}.xml"
+
+    def create_assessment_item(
+        self, assessment_item, processed_data: Dict[str, Any]
+    ) -> tuple[str, bytes]:
+        """Create QTI assessment item XML."""
+
+        # Skip Perseus questions as they can't be easily converted
+        if assessment_item.type == exercises.PERSEUS_QUESTION:
+            raise ValueError(
+                f"Perseus questions are not supported in QTI format: {assessment_item.assessment_id}"
+            )
+
+        if assessment_item.type in choice_interactions:
+            (
+                interaction,
+                response_declaration,
+            ) = self._create_choice_interaction_and_response(processed_data)
+        elif assessment_item.type in text_entry_interactions:
+            (
+                interaction,
+                response_declaration,
+            ) = self._create_text_entry_interaction_and_response(processed_data)
+        else:
+            raise ValueError(f"Unsupported question type: {assessment_item.type}")
+
+        # Create item body with the interaction
+        item_body = ItemBody(children=[interaction])
+
+        # Create outcome declaration
+        outcome_declaration = OutcomeDeclaration(
+            identifier="SCORE", cardinality=Cardinality.SINGLE, base_type=BaseType.FLOAT
+        )
+
+        # Create response processing
+        response_processing = ResponseProcessing(
+            template="https://purl.imsglobal.org/spec/qti/v3p0/rptemplates/match_correct"
+        )
+
+        language = (
+            self.ccnode.language.lang_code
+            if self.ccnode.language
+            else self.default_language
+        )
+
+        # Create the assessment item
+        qti_item = AssessmentItem(
+            identifier=f"item_{assessment_item.assessment_id}",
```

Comment:
I have realized that this is not terribly helpful to add this prefix. I will update to simply use the assessment_id.

Reply:
This won't work either, because identifiers are required to be 32 characters or less. Instead we'll encode them as base64 and prefix with a single letter.